src/libstd/sys/unix/fs.rs

```diff
@@ -915,21 +933,20 @@ pub fn copy(from: &Path, to: &Path) -> io::Result<u64> {
                 match err.raw_os_error() {
                     Some(os_err) if os_err == libc::ENOSYS
                                  || os_err == libc::EXDEV
+                                 || os_err == libc::EINVAL
                                  || os_err == libc::EPERM => {
                         // Try fallback io::copy if either:
                         // - Kernel version is < 4.5 (ENOSYS)
                         // - Files are mounted on different fs (EXDEV)
                         // - copy_file_range is disallowed, for example by seccomp (EPERM)
+                        // - source or destination cannot be used with copy_file_range()
```

Comment:
The others definitely make sense to me but is `EINVAL` something that legitimately comes up? That seems sort of like a big hammer to disallow such a wide range of possible errors causing them to stop using the more optimized pass.

In any case could the documentation here be updated to indicate which error is associated with this case?